tests/test_class_attributes.rs

```diff
@@ -0,0 +1,29 @@
+use pyo3::prelude::*;
+
+mod common;
+
+#[pyclass]
+struct Foo { }
+
+#[pymethods]
+impl Foo {
+    #[attribute]
+    fn foo() -> i32 {
+        5
+    }
+
+    #[attribute]
+    #[name = "BAR"]
+    fn bar() -> String {
+        "bar".to_string()
+    }
+}
+
+#[test]
```

Comment:
Mentioned this in my comment, but adding an in-line note so it doesn't get lost: I think it would be a good idea to add tests for the error cases, to ensure that they raise the correct exception.

Could be done here or it could be done in the `examples/rustapi_module` tests, which are written in Python and where you can use `pytest.raises`.